### PR TITLE
Small import optimizations

### PR DESCRIPTION
**Pull request by Scienfitz**
- moves polytope sampler import from botorch utils to the function where its needed (saves some 20% of import time)

One of the things I notice is it seems that our entire utils module is loaded whenever you import a utility from there
that doesnt seem optimal as some of the utils also use torch and such things, do we wanna change that? one aspect of our design was that you can import anything form utils top level instead of the exact submodule but that seems to hinder cleanly separated loading, any opinions?

### CODE REVIEW

**Comment by Scienfitz** [comment]
@AdrianSosic please stop making PR comments and make threads instead :D 

I've pushed a proposal commit for removing the high level imports from utils
However, it seems to have a miniscule effect (0.1s savings). A quick look suggests that the sklearn import (in sampling algos) thats not done anymore depends on lots of scipy stuff, however, that gets imported anyway in another arch by botorch, resulting in not much savings.

looking at tuna reports theres not much more stuff to reduce, except torch and botorch.

I wouldnt touch any of the torch stuff in this PR as maybe rjavadi already is looking into this.

**Comment by AdrianSosic** [comment]
🤣 you are telling me not to make PR comments .... from your own PR comment 🤣  
So this is the last one, just because I can 😎 
But PR looking good to me, you have my blessing to merge 🚀

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,23 +13,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - Optional `chem` dependencies are lazily imported
 - Order of README sections
+- Imports from top level `baybe.utils` no longer possible
```
Comment (AdrianSosic) [inline CHANGELOG.md:16]: Perhaps we could use this opportunity to also rename `utils.numeric` to `utils.numerical` for concistency?

Comment (Scienfitz) [reply CHANGELOG.md:16]: done

**Review by AVHopp — APPROVED** [review]

**Pull request merged**